From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Christian Hewitt <christianshewitt@gmail.com>
Date: Wed, 16 Feb 2022 07:27:07 +0000
Subject: dt-bindings: arm: amlogic: add support for Radxa Zero2

The Radxa Zero2 is a small form-factor SBC using the Amlogic
A311D chip.

Signed-off-by: Christian Hewitt <christianshewitt@gmail.com>
Signed-off-by: Yuntian Zhang <yt@radxa.com>
---
 arch/arm64/boot/dts/amlogic/meson-g12b-radxa-zero2.dts | 119 ++++++++--
 1 file changed, 103 insertions(+), 16 deletions(-)

diff --git a/arch/arm64/boot/dts/amlogic/meson-g12b-radxa-zero2.dts b/arch/arm64/boot/dts/amlogic/meson-g12b-radxa-zero2.dts
index 890f5bfebb03..e261ba2a4b47 100644
--- a/arch/arm64/boot/dts/amlogic/meson-g12b-radxa-zero2.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-g12b-radxa-zero2.dts
@@ -17,7 +17,7 @@
 
 / {
 	compatible = "radxa,zero2", "amlogic,a311d", "amlogic,g12b";
-	model = "Radxa Zero2";
+	model = "Radxa Zero 2";
 
 	aliases {
 		serial0 = &uart_AO;
@@ -28,6 +28,13 @@ chosen {
 		stdout-path = "serial0:115200n8";
 	};
 
+   fan0: pwm-fan {
+		compatible = "pwm-fan";
+		#cooling-cells = <2>;
+		cooling-levels = <0 64 128 192 255>;
+		pwms = <&pwm_AO_ab 0 40000 0>;
+	};
+
 	memory@0 {
 		device_type = "memory";
 		reg = <0x0 0x0 0x0 0x80000000>;
@@ -54,6 +61,17 @@ led-green {
 		};
 	};
 
+	cvbs-connector {
+		status = "disabled";
+		compatible = "composite-video-connector";
+
+		port {
+			cvbs_connector_in: endpoint {
+				remote-endpoint = <&cvbs_vdac_out>;
+			};
+		};
+	};
+
 	hdmi-connector {
 		compatible = "hdmi-connector";
 		type = "a";
@@ -77,7 +95,15 @@ sdio_pwrseq: sdio-pwrseq {
 		clock-names = "ext_clock";
 	};
 
-	ao_5v: regulator-ao-5v {
+	typec2_vbus: regulator-typec2_vbus {
+		compatible = "regulator-fixed";
+		regulator-name = "TYPEC2_VBUS";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		vin-supply = <&ao_5v>;
+	};
+
+	ao_5v: regulator-ao_5v {
 		compatible = "regulator-fixed";
 		regulator-name = "AO_5V";
 		regulator-min-microvolt = <5000000>;
@@ -85,7 +111,7 @@ ao_5v: regulator-ao-5v {
 		regulator-always-on;
 	};
 
-	vcc_1v8: regulator-vcc-1v8 {
+	vcc_1v8: regulator-vcc_1v8 {
 		compatible = "regulator-fixed";
 		regulator-name = "VCC_1V8";
 		regulator-min-microvolt = <1800000>;
@@ -94,7 +120,7 @@ vcc_1v8: regulator-vcc-1v8 {
 		regulator-always-on;
 	};
 
-	vcc_3v3: regulator-vcc-3v3 {
+	vcc_3v3: regulator-vcc_3v3 {
 		compatible = "regulator-fixed";
 		regulator-name = "VCC_3V3";
 		regulator-min-microvolt = <3300000>;
@@ -104,7 +130,7 @@ vcc_3v3: regulator-vcc-3v3 {
 		/* FIXME: actually controlled by VDDCPU_B_EN */
 	};
 
-	vddao_1v8: regulator-vddao-1v8 {
+	vddao_1v8: regulator-vddao_1v8 {
 		compatible = "regulator-fixed";
 		regulator-name = "VDDIO_AO1V8";
 		regulator-min-microvolt = <1800000>;
@@ -113,7 +139,7 @@ vddao_1v8: regulator-vddao-1v8 {
 		regulator-always-on;
 	};
 
-	vddao_3v3: regulator-vddao-3v3 {
+	vddao_3v3: regulator-vddao_3v3 {
 		compatible = "regulator-fixed";
 		regulator-name = "VDDAO_3V3";
 		regulator-min-microvolt = <3300000>;
@@ -176,6 +202,7 @@ sound {
 		assigned-clock-rates = <294912000>,
 				       <270950400>,
 				       <393216000>;
+		status = "okay";
 
 		dai-link-0 {
 			sound-dai = <&frddr_a>;
@@ -214,7 +241,7 @@ codec {
 		};
 	};
 
-	wifi32k: clock-0 {
+	wifi32k: wifi32k {
 		compatible = "pwm-clock";
 		#clock-cells = <0>;
 		clock-frequency = <32768>;
@@ -222,6 +249,18 @@ wifi32k: clock-0 {
 	};
 };
 
+&periphs_pinctrl {
+	/* Ensure the TYPE C controller irq pin is not driven by the SoC */
+	fusb302_irq_pins: fusb302_irq {
+		mux {
+			groups = "GPIOA_13";
+			function = "gpio_periphs";
+			bias-pull-up;
+			output-disable;
+		};
+	};
+};
+
 &arb {
 	status = "okay";
 };
@@ -286,6 +325,12 @@ &cpu103 {
 	clock-latency = <50000>;
 };
 
+&cvbs_vdac_port {
+	cvbs_vdac_out: endpoint {
+		remote-endpoint = <&cvbs_connector_in>;
+	};
+};
+
 &frddr_a {
 	status = "okay";
 };
@@ -302,7 +347,7 @@ &gpio {
 	gpio-line-names =
 		/* GPIOZ */
 		"PIN_27", "PIN_28", "PIN_7", "PIN_11", "PIN_13", "PIN_15", "PIN_18", "PIN_40",
-		"", "", "", "", "", "", "", "",
+		"PIN_16", "PIN_22", "", "", "", "", "", "",
 		/* GPIOH */
 		"", "", "", "", "PIN_19", "PIN_21", "PIN_24", "PIN_23",
 		"",
@@ -310,10 +355,10 @@ &gpio {
 		"", "", "", "", "", "", "", "",
 		"", "", "", "", "EMMC_PWRSEQ", "", "", "",
 		/* GPIOC */
-		"", "", "", "", "", "", "SD_CD", "PIN_36",
+		"", "", "", "", "", "", "SD_CD", "TYPEC_MUX",
 		/* GPIOA */
-		"PIN_32", "PIN_12", "PIN_35", "", "", "PIN_38", "", "",
-		"", "", "", "", "LED_GREEN", "PIN_31", "PIN_3", "PIN_5",
+		"PIN_32", "PIN_12", "PIN_35", "PIN_36", "PIN_31", "PIN_38", "", "",
+		"", "", "", "", "LED_GREEN", "FUSB_IRQ", "PIN_3", "PIN_5",
 		/* GPIOX */
 		"", "", "", "", "", "", "SDIO_PWRSEQ", "",
 		"", "", "", "", "", "", "", "",
@@ -324,7 +369,7 @@ &gpio_ao {
 	gpio-line-names =
 		/* GPIOAO */
 		"PIN_8", "PIN_10", "", "BTN_POWER", "", "", "", "PIN_29",
-		"PIN_33", "PIN_37", "FAN", "",
+		"PIN_33", "PIN_37", "", "FAN",
 		/* GPIOE */
 		"", "", "";
 };
@@ -342,12 +387,50 @@ hdmi_tx_tmds_out: endpoint {
 	};
 };
 
+&cpu_thermal {
+	cooling-maps {
+		map0 {
+			trip = <&cpu_passive>;
+			cooling-device = <&fan0 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+		};
+	};
+};
+
+&ddr_thermal {
+	cooling-maps {
+		map0 {
+			trip = <&ddr_passive>;
+			cooling-device = <&fan0 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+		};
+	};
+};
+
 &ir {
 	status = "disabled";
 	pinctrl-0 = <&remote_input_ao_pins>;
 	pinctrl-names = "default";
 };
 
+&i2c3 {
+	pinctrl-0 = <&i2c3_sda_a_pins>, <&i2c3_sck_a_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+
+	fusb302@22 {
+		compatible = "fcs,fusb302";
+		reg = <0x22>;
+
+		pinctrl-0 = <&fusb302_irq_pins>;
+		pinctrl-names = "default";
+		interrupt-parent = <&gpio_intc>;
+		interrupts = <74 IRQ_TYPE_LEVEL_LOW>;
+
+		vbus-supply = <&typec2_vbus>;
+
+		status = "okay";
+	};
+};
+
 &pwm_ab {
 	pinctrl-0 = <&pwm_a_e_pins>;
 	pinctrl-names = "default";
@@ -360,7 +443,7 @@ &pwm_ef {
 	pinctrl-0 = <&pwm_e_pins>;
 	pinctrl-names = "default";
 	clocks = <&xtal>;
-	clock-names = "clkin0";
+	clock-names = "clkin2";
 	status = "okay";
 };
 
@@ -368,7 +451,7 @@ &pwm_AO_ab {
 	pinctrl-0 = <&pwm_ao_a_pins>;
 	pinctrl-names = "default";
 	clocks = <&xtal>;
-	clock-names = "clkin0";
+	clock-names = "clkin3";
 	status = "okay";
 };
 
@@ -376,7 +459,7 @@ &pwm_AO_cd {
 	pinctrl-0 = <&pwm_ao_d_e_pins>;
 	pinctrl-names = "default";
 	clocks = <&xtal>;
-	clock-names = "clkin1";
+	clock-names = "clkin4";
 	status = "okay";
 };
 
@@ -396,7 +479,7 @@ &sd_emmc_a {
 
 	bus-width = <4>;
 	cap-sd-highspeed;
-	max-frequency = <100000000>;
+	max-frequency = <80000000>;
 
 	non-removable;
 	disable-wp;
@@ -487,3 +570,7 @@ &uart_AO {
 &usb {
 	status = "okay";
 };
+
+&usb3_pcie_phy {
+	phy-supply = <&typec2_vbus>;
+};
-- 
Armbian

